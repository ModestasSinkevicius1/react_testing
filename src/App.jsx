import logo from './logo.svg';
import './App.css';
import Greeter from './components/Greeter/Greeter';
import List from './components/List/List';
import { useState } from 'react';
import randColor from './Utilities/random_color.js'
import List2 from './components/ListFromGet/List2';

function App() {

  const [records, serRecords] = useState([...Array(50)].map(a => ({ name: randColor() }) ));

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
        <span className='home'>Hello</span>
        <span style={{opacity: 0}}>Bye</span>
        <List2 />
        <Greeter />
        <List records={records}/>
      </header>
    </div>
  );
}

export default App;
