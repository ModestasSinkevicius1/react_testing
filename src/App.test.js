import { render, screen } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('check if my text renders', () => {
  render(<App />);
  const linkElement = screen.getByText('Hello');
  //screen.debug(linkElement);
  expect(linkElement).toBeInTheDocument();
});

test("bye is not visible", () => {
  // eslint-disable-next-line no-unused-vars
  render(<App />);
  const hw = screen.getByText("Bye");
  expect(hw).not.toBeVisible();
});

test("check if home has 'home' class", () => {
  const { container } = render(<App />);
  // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
  const hw = container.querySelector('.home')
  expect(hw).toBeInTheDocument();
});

test("check if text is formed corectly with tags", () => {
  render(<App />);
  const text = screen.getByText((content, node) => node.textContent === 'Edit src/App.js and save to reload.')
  expect(text).toBeInTheDocument();
});

// test('Match snapshot', () => {
//   const { container } = render(<App />)

//   expect(container).toMatchSnapshot();
// });

