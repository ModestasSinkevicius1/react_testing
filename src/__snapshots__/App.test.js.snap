// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      <span
        class="home"
      >
        Hello
      </span>
      <span
        style="opacity: 0;"
      >
        Bye
      </span>
      <div
        class="greeter"
      >
        <label
          for="input_text"
        >
          Name:
        </label>
        <input
          class="input-text"
          id="input_text"
          placeholder="Type name"
          type="text"
          value=""
        />
        <button
          class="btn"
        >
          Show name
        </button>
      </div>
      <div
        class="list"
      >
        <h2>
          Random color codes
        </h2>
        <ul>
          <li
            class="item"
          >
            #97ee3c
          </li>
          <li
            class="item"
          >
            #18ad03
          </li>
          <li
            class="item"
          >
            #1f510d
          </li>
          <li
            class="item"
          >
            #24429e
          </li>
          <li
            class="item"
          >
            #43a1c8
          </li>
          <li
            class="item"
          >
            #e32134
          </li>
          <li
            class="item"
          >
            #b5746d
          </li>
          <li
            class="item"
          >
            #3b8aa
          </li>
          <li
            class="item"
          >
            #ed2269
          </li>
          <li
            class="item"
          >
            #6b79ce
          </li>
          <li
            class="item"
          >
            #cdf7ef
          </li>
          <li
            class="item"
          >
            #ffe505
          </li>
          <li
            class="item"
          >
            #938f76
          </li>
          <li
            class="item"
          >
            #e8ea62
          </li>
          <li
            class="item"
          >
            #a494e9
          </li>
          <li
            class="item"
          >
            #dff13c
          </li>
          <li
            class="item"
          >
            #641ed7
          </li>
          <li
            class="item"
          >
            #f78efd
          </li>
          <li
            class="item"
          >
            #98a13
          </li>
          <li
            class="item"
          >
            #48fc1
          </li>
          <li
            class="item"
          >
            #f15f13
          </li>
          <li
            class="item"
          >
            #e35127
          </li>
          <li
            class="item"
          >
            #deb2a
          </li>
          <li
            class="item"
          >
            #6aa0da
          </li>
          <li
            class="item"
          >
            #cdd68c
          </li>
          <li
            class="item"
          >
            #fb10bf
          </li>
          <li
            class="item"
          >
            #3e55e7
          </li>
          <li
            class="item"
          >
            #3b2e8d
          </li>
          <li
            class="item"
          >
            #87498
          </li>
          <li
            class="item"
          >
            #f23134
          </li>
          <li
            class="item"
          >
            #734710
          </li>
          <li
            class="item"
          >
            #421d73
          </li>
          <li
            class="item"
          >
            #221848
          </li>
          <li
            class="item"
          >
            #a271cc
          </li>
          <li
            class="item"
          >
            #55eca7
          </li>
          <li
            class="item"
          >
            #e8e65b
          </li>
          <li
            class="item"
          >
            #56ba0b
          </li>
          <li
            class="item"
          >
            #b6191d
          </li>
          <li
            class="item"
          >
            #b819cf
          </li>
          <li
            class="item"
          >
            #d56932
          </li>
          <li
            class="item"
          >
            #58665e
          </li>
          <li
            class="item"
          >
            #8fb441
          </li>
          <li
            class="item"
          >
            #c0bcf3
          </li>
          <li
            class="item"
          >
            #5a2d81
          </li>
          <li
            class="item"
          >
            #5a23cd
          </li>
          <li
            class="item"
          >
            #e275fc
          </li>
          <li
            class="item"
          >
            #daf781
          </li>
          <li
            class="item"
          >
            #1e5092
          </li>
          <li
            class="item"
          >
            #c30587
          </li>
          <li
            class="item"
          >
            #d68677
          </li>
        </ul>
      </div>
    </header>
  </div>
</div>
`;
